Services/Style/classes/Setup/class.ilStyleDBUpdateSteps.php

```diff
@@ -99,4 +99,105 @@ public function step_5()
     {
         $this->db->renameTableColumn('style_char', "deprecated", 'outdated');
     }
+
+    public function step_6()
+    {
+        if (!$this->db->tableExists('sty_rep_container')) {
+            $fields = [
+                'ref_id' => [
+                    'type' => 'integer',
+                    'length' => 4,
+                    'notnull' => true,
+                    'default' => 0
+                ],
+                'reuse' => [
+                    'type' => 'integer',
+                    'length' => 1,
+                    'notnull' => true,
+                    'default' => 0
+                ]
+            ];
+
+            $this->db->createTable('sty_rep_container', $fields);
+            $this->db->addPrimaryKey('sty_rep_container', ['ref_id']);
+        }
+    }
+
+    public function step_7()
+    {
+        $set = $this->db->queryF(
+            "SELECT * FROM content_object ",
+            [],
+            []
+        );
+        while ($rec = $this->db->fetchAssoc($set)) {
+            $this->db->replace(
+                "style_usage",
+                array(
+                    "obj_id" => array("integer", (int) $rec["id"])
+                ),
+                array(
+                    "style_id" => array("integer", (int) $rec["stylesheet"])
+                )
+            );
+        }
+    }
+
+    public function step_8()
+    {
+        $set = $this->db->queryF(
+            "SELECT * FROM content_page_data ",
+            [],
+            []
+        );
+        while ($rec = $this->db->fetchAssoc($set)) {
+            $this->db->replace(
+                "style_usage",
+                array(
+                    "obj_id" => array("integer", (int) $rec["content_page_id"])),
+                array(
+                    "style_id" => array("integer", (int) $rec["stylesheet"]))
+            );
+        }
```

Comment:
Should't we drop the table column after the migration?

Reply:
Yes, but how? I don't think we want the Style service to modify the Content Page tables, right? Is it possible to define a dependency in a component DB Update step on a step in another component? We must ensure the right sequence, though.

Reply:
Yes. this is possible, but difficult :-). Maybe we could have a chat about this (VC :-)) and discuss, if this is really feasible?

Reply:
What would be your preference? Another option would be to keep the field and mark it in the Content Page Roadmap for deletion with ILIAS 9.

Reply:
I will mark it in the ROADMAP file. The same applies for the upcoming forum changes.

If you are still interested in the `Dependency` handling, call me :-).